app/src/main/java/com/example/drawingApp/MainActivity.kt

```diff
@@ -1,23 +1,111 @@
 package com.example.drawingApp
 
 import android.os.Bundle
+import android.os.StrictMode
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
 import androidx.activity.viewModels
+import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.widget.addTextChangedListener
 
 class MainActivity : AppCompatActivity() {
+    private val MAX_CHAR_LEN = 16
     private val model: MainViewModel by viewModels()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-
+        enableStrictMode()
         val colorCircleView = findViewById<SelectedColorView>(R.id.colorCircle)
         val drawingFieldView = findViewById<DrawingFieldView>(R.id.drawField)
 
+        drawingFieldView.onFinishLine = {
+            model.activeBitmap = it
+        }
+        findViewById<TextView>(R.id.newLayer).setOnClickListener {
+            showLayerAlertDialog()
+        }
+
+        model.activeBitmap?.let {
+            drawingFieldView.setBitmap(it)
+            drawingFieldView.invalidate()
+        }
+
+
+
```

Comment:
Extra space

Reply:
removed, formatting doesn't get rid of extra space I suppose.